AstarteDeviceSDKCSharp/Data/AstarteFailedMessageStorage.cs

```diff
@@ -365,6 +365,64 @@ public async Task DeleteByGuidAsync(Guid applicationMessageId)
             }
         }
 
+        public async Task MarkAsProcessed(Guid applicationMessageId)
+        {
+
+            try
+            {
+                if (sqliteReadConnection.State != ConnectionState.Open)
+                {
+                    sqliteReadConnection = new SqliteConnection(_dbConnectionString);
+                    sqliteReadConnection.Open();
+                }
+
+                SqliteCommand cmd = new SqliteCommand(
+                    "UPDATE AstarteFailedMessages SET processed = 1 WHERE guid = @guid ;", sqliteReadConnection);
+
+                cmd.Parameters.AddWithValue("@guid", applicationMessageId);
+                await cmd.ExecuteNonQueryAsync();
+                cmd.Dispose();
+
+                if (_astarteFailedMessageVolatile is not null)
+                {
+                    var messageVolatile = _astarteFailedMessageVolatile
+                        .Where(x => x.Guid == applicationMessageId)
+                        .FirstOrDefault();
+                    if (messageVolatile is not null)
+                    {
+                        _astarteFailedMessageVolatile.Remove(messageVolatile);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Trace.WriteLine($"Failed to delete fallback message from database. Error message: {ex.Message}");
```

Comment:
I think this message is swapped from the `DeleteProcessed` method.
```suggestion
                Trace.WriteLine($"Failed to mark processed fallback message in database. Error message: {ex.Message}");
```

Reply:
Changed.